### ISSUE TITLE
Cross-referencing summary tables generated by qwraps2::summary_table()

### ISSUE THREAD

**Issue by e-linc** [dewittpe/qwraps2#84]
I was trying to cross-reference a summary table created using `qwraps2::summary_table(`) in my R Markdown Word report, but it did not work with my .Rmd file below:

    ---
    title: "Summary table in R Markdown"
    output: bookdown::word_document2
    ---
    
    ```{r setup, include=FALSE}
    knitr::opts_chunk$set(echo = FALSE)
    ```
    
    A summary of the group is in Table \@ref(tab:summ-tab):
    
    ```{r summ-tab, results="asis"}
    
    library(tidyverse)
    library(qwraps2)
    options(qwraps2_markup = "markdown")
    
    tdf <- tibble(
      age = runif(n = 30, min = 20, max = 40),
      sex = sample(c("M", "F"), size = 30, replace = TRUE),
      bmi = runif(n = 30, min = 15, max = 45),
      disease = sample(1:2, size = 30, replace = TRUE)
    )
    
    summary_str <- list(
      "Sex, n (%)" = list(
        "Male" = ~ n_perc0(sex == "M", digits = 1),
        "Female" = ~ n_perc0(sex == "F", digits = 1)
      ),
      "Age, mean &plusmn; SD" = list(
        "Age (yr)" = ~ mean_sd(age, digits = 0)
      ),
      "BMI, mean &plusmn; SD" = list(
        "BMI (kg m^-2^)" = ~ mean_sd(bmi, digits = 1)
      )
    )

    tdf %>%
      group_by(disease) %>% 
      summary_table(summary_str) %>% 
      print()      
      # knitr::kable(caption = "Personal summary")
      # qable(caption = "Personal summary")
    
    ```

As shown in the screenshot below, the `print()` method formatted the table correctly but there was no cross-reference in the text and I am not how to add a caption to the table:

![qwraps2_print](https://user-images.githubusercontent.com/26250149/71904594-5589c300-315e-11ea-9bb8-d78e3eae66d8.png)

For my other tables in the report I used `knitr::kable()`, which would generate a table label for cross-referencing, so I tried to output the table using `knitr::kable()` instead of `print()`. The cross-reference worked, but the table itself was not rendered in my .docx file:

![qwraps2_kable](https://user-images.githubusercontent.com/26250149/71904716-9255ba00-315e-11ea-8682-d66031f11b8a.png)

Considering that `qwraps2::qable()` is a wrapper for `knitr::kable()`, I thought it would be able to generate a label as `kable()`, and render the table correctly, but the cross-reference was broken again, and it only half-rendered the table (missing the group titles):

![qwraps2_qable](https://user-images.githubusercontent.com/26250149/71904860-d8128280-315e-11ea-910d-e527173c9a95.png)

I searched online but did not find a solution to any question similar to mine. I asked [the same question](https://stackoverflow.com/questions/59580332/cross-referencing-summary-tables-by-qwraps2summary-table) on Stack Overflow, but have not got an answer. Am I doing something unusual or wrong?

**Comment by dewittpe** [comment]
Digging through `knitr::kable` and qwraps2 I think I have found the issue.  In qwraps2 `format` is either latex or markdown but in `knitr::kable` format also refers to the target output.  I will need to revisit how qwraps2 uses the format argument and how it is passed through to knitr functions.

My work flow has generally relied on explicit writing of the table references for tables and figures in markdown documents.  qwraps2 was written with that in mind.  This issue is a good example for way I need to revisit how qwraps2 works.  Thanks for posting this.  I'll post updates here and on SO.

**Comment by e-linc** [comment]
> My work flow has generally relied on explicit writing of the table references for tables and figures in markdown documents. qwraps2 was written with that in mind.

Thanks very much for your reply. I am fairly new to R Markdown reporting. Would you mind sharing how you explicitly cross-referenced `qwraps2` tables? Due to the issue reported here, I am manually modifying all cross-references to tables in the Word output. Many thanks again!

**Comment by dewittpe** [comment]
@e-clin I have finally had some time to work on this package and I've got a fix for this problem that will be released in v0.6.0 soon.   For the moment you can get the development version of the package from github, or wait a couple more days before I submit to v0.6.0 to CRAN.

The fix to this issue is the addition of the arguments `qable_args` and `kable_args` to the `summary_table` call.  These additional arguments help move argument values from `summary_table` to `qable` and ultimately to `knitr::kable`

Here is an example of an .Rmd file and a screen shot of the resulting .docx file after calling `rmarkdown::render` on the .Rmd file.

    ---
    title: "Summary table in R Markdown"
    output: bookdown::word_document2
    ---
    
    ```{r setup, include=FALSE}
    library(qwraps2)
    options(qwraps2_markup = "markdown")
    ```
    
    ```{r label = "summ_tab_build", include = FALSE}
    tdf <- data.frame(
      age = runif(n = 30, min = 20, max = 40),
      sex = sample(c("M", "F"), size = 30, replace = TRUE),
      bmi = runif(n = 30, min = 15, max = 45),
      disease = sample(1:2, size = 30, replace = TRUE)
    )
    
    summary_str <- list(
      "Sex, n (%)" = list(
        "Male" = ~ n_perc0(sex == "M", digits = 1),
        "Female" = ~ n_perc0(sex == "F", digits = 1)
      ),
      "Age, mean &plusmn; SD" = list(
        "Age (yr)" = ~ mean_sd(age, digits = 0)
      ),
      "BMI, mean &plusmn; SD" = list(
        "BMI (kg m^-2^)" = ~ mean_sd(bmi, digits = 1)
      )
    )
    ```
    
    A summary of the the data by disease group is in Table \@ref(tab:stab) and is
    generated via `qwraps2::summary_table`.
    More text goes here ...  Table \@ref(tab:stab2) uses `knitr::kable` directly.
    
    ```{r label = "stab", echo = FALSE, results = "asis"}
    summary_table(x = tdf, summaries = summary_str, by = "disease", qable_args = list(kable_args = list(caption = "Personal summary")))
    ```
    
    More text goes here.
    
    ```{r label = 'stab2', echo = FALSE, results = "asis"}
    knitr::kable(tdf, caption = "simple table")
    ```
    
    More text goes here.



![image](https://github.com/dewittpe/qwraps2/assets/3136617/97a05d8a-42df-410a-839d-8ec69e3ebe76)
